sys/dev/misc/efirt/efidev.c

```diff
@@ -77,16 +77,31 @@ efidev_ioctl(struct dev_ioctl_args *ap)
 	case EFIIOC_GET_TABLE:
 	{
 		struct efi_get_table_ioc *egtioc =
-		    (struct efi_get_table_ioc *)addr;
+			(struct efi_get_table_ioc *)addr;
+		
+		void *data;
 
 		error = efi_get_table(&egtioc->uuid, &egtioc->ptr);
+
+		struct efi_esrt_table *esrt = NULL;
+		esrt = egtioc->ptr;
+
+		kprintf("IOCTL: esrt->fw_resource_count = %d\n", esrt->fw_resource_count);
+
+		egtioc->table_len = sizeof(*esrt)+ (sizeof(struct efi_esrt_entry_v1) * esrt->fw_resource_count);
+		data = kmalloc(egtioc->table_len, M_TEMP, M_WAITOK);
+
+		kprintf("table_len = %ln\n", &egtioc->table_len);
+
+		error = copyout(data, &egtioc->ptr, egtioc->table_len);
```

Comment:
@SergiiDmytruk can you look here? it is a more clean PR with some updates. With that modifications, I got `-1` error from copyout. `esrt->fw_resource_count` value is read correctly, the only problem is passing ESRT values to user space. Can you give me some tips on how to fix that?

Reply:
If `egtioc->ptr` is NULL or `egtioc->buf_len < egtioc->table_len`, you shouldn't copy anything. And case of `egtioc->ptr ==  NULL` should return "success" (0) as it's a request for table size.

Reply:
@SergiiDmytruk I end with copyout error - `EFAULT 14 Bad address`

Reply:
I guess `efi_get_table(&egtioc->uuid, &egtioc->ptr);` was the error here, so I assume things work now.

Reply:
@SergiiDmytruk sure, here is coninuation: https://github.com/3mdeb/DragonFlyBSD/pull/6